proparse/src/main/java/org/prorefactor/treeparser/TreeParserException.java

```diff
@@ -1,17 +1,16 @@
 package org.prorefactor.treeparser;
 
-public class TreeParserException extends Exception {
-  private static final long serialVersionUID = 4003548635999211002L;
+import antlr.SemanticException;
+
+public class TreeParserException extends SemanticException {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This class has 6 parents which is greater than 5 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)